dev-packages/cli/src/download-plugins.ts

```diff
@@ -77,6 +77,9 @@ export default async function downloadPlugins(options: DownloadPluginsOptions =
         temp.cleanupSync();
     }
     failures.forEach(console.error);
+    if (failures) {
```

Comment:
```suggestion
    if (failures.length) {
```

The current condition produces false negatives. To verify you can:
1. keep `theiaPlugins` as default:
https://github.com/eclipse-theia/theia/blob/055175566b279cded090f634c6672d17e067f904/package.json#L85
2. execute `yarn download:plugins`
3. the script will download all plugins successfully but still output an error:

<details>
<summary>Output</summary>

```
$ theia download:plugins
--- downloading plugins ---
+ vscode-builtin-clojure: downloaded successfully 
+ vscode-builtin-bat: downloaded successfully 
+ vscode-builtin-coffeescript: downloaded successfully 
+ vscode-builtin-configuration-editing: downloaded successfully 
+ vscode-builtin-csharp: downloaded successfully 
+ vscode-builtin-css: downloaded successfully 
+ vscode-builtin-docker: downloaded successfully 
+ vscode-builtin-debug-auto-launch: downloaded successfully 
+ vscode-builtin-fsharp: downloaded successfully 
+ vscode-builtin-go: downloaded successfully 
+ vscode-builtin-cpp: downloaded successfully 
+ vscode-builtin-groovy: downloaded successfully 
+ vscode-builtin-hlsl: downloaded successfully 
+ vscode-builtin-grunt: downloaded successfully 
+ vscode-builtin-gulp: downloaded successfully 
+ vscode-builtin-ini: downloaded successfully 
+ vscode-builtin-html: downloaded successfully 
+ vscode-builtin-css-language-features: downloaded successfully 
+ vscode-builtin-java: downloaded successfully 
+ vscode-builtin-json: downloaded successfully 
+ vscode-builtin-handlebars: downloaded successfully 
+ vscode-builtin-jake: downloaded successfully 
+ vscode-builtin-javascript: downloaded successfully 
+ vscode-builtin-log: downloaded successfully 
+ vscode-builtin-lua: downloaded successfully 
+ vscode-builtin-markdown: downloaded successfully 
+ vscode-builtin-make: downloaded successfully 
+ vscode-builtin-emmet: downloaded successfully 
+ vscode-builtin-less: downloaded successfully 
+ vscode-builtin-merge-conflict: downloaded successfully 
+ vscode-builtin-perl: downloaded successfully 
+ vscode-builtin-powershell: downloaded successfully 
+ vscode-builtin-pug: downloaded successfully 
+ vscode-builtin-objective-c: downloaded successfully 
+ vscode-builtin-npm: downloaded successfully 
+ vscode-builtin-razor: downloaded successfully 
+ vscode-builtin-python: downloaded successfully 
+ vscode-builtin-rust: downloaded successfully 
+ vscode-builtin-scss: downloaded successfully 
+ vscode-builtin-node-debug: downloaded successfully 
+ vscode-builtin-json-language-features: downloaded successfully 
+ vscode-builtin-shaderlab: downloaded successfully 
+ vscode-builtin-shellscript: downloaded successfully 
+ vscode-builtin-js-debug: downloaded successfully 
+ vscode-builtin-sql: downloaded successfully 
+ vscode-builtin-theme-abyss: downloaded successfully 
+ vscode-builtin-markdown-language-features: downloaded successfully 
+ vscode-builtin-r: downloaded successfully 
+ vscode-builtin-theme-kimbie-dark: downloaded successfully 
+ vscode-builtin-theme-defaults: downloaded successfully 
+ vscode-builtin-theme-monokai: downloaded successfully 
+ vscode-builtin-theme-monokai-dimmed: downloaded successfully 
+ vscode-builtin-theme-quietlight: downloaded successfully 
+ vscode-builtin-theme-red: downloaded successfully 
+ vscode-builtin-swift: downloaded successfully 
+ vscode-builtin-theme-solarized-dark: downloaded successfully 
+ vscode-builtin-node-debug2: downloaded successfully 
+ vscode-builtin-theme-tomorrow-night-blue: downloaded successfully 
+ vscode-builtin-vb: downloaded successfully 
+ vscode-builtin-typescript: downloaded successfully 
+ vscode-builtin-xml: downloaded successfully 
+ vscode-builtin-yaml: downloaded successfully 
+ vscode-builtin-html-language-features: downloaded successfully 
+ vscode-builtin-icon-theme-seti: downloaded successfully 
+ vscode-editorconfig: downloaded successfully 
+ vscode-eslint: downloaded successfully 
+ vscode-references-view: downloaded successfully 
+ vscode-builtin-typescript-language-features: downloaded successfully 
+ vscode-builtin-ruby: downloaded successfully 
Error: Errors downloading plugins
    at Object.<anonymous> (/home/evinfug/workspaces/theia-work/dev-packages/cli/lib/download-plugins.js:114:31)
    at step (/home/evinfug/workspaces/theia-work/dev-packages/cli/lib/download-plugins.js:59:23)
    at Object.next (/home/evinfug/workspaces/theia-work/dev-packages/cli/lib/download-plugins.js:40:53)
    at fulfilled (/home/evinfug/workspaces/theia-work/dev-packages/cli/lib/download-plugins.js:31:58)
error Command failed with exit code 1.
```

</details>

Reply:
Just make it `failures.length > 0` to fully express the intent.

Reply:
Oh of course ... thank you ! I have updated the branch with the suggested change.

Thank you also for linking with the issue. I did not realise this was already reported.

Reply:
Actually on the screenshot I posted it was a false negative. "unsupported file types" are not reported as failures. I suggest https://github.com/eclipse-theia/theia/pull/8797 for this